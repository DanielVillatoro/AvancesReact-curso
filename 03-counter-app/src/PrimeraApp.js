import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
// import React, { Fragment } from 'react';
//FC

//<fragment>
const PrimeraApp = ({saludo,subtitulo}) => {

    return (
        <Fragment>  
            <h1>{saludo}</h1>
            {/* <pre>{JSON.stringify(saludo,null,3)}</pre> */}
            <p>{subtitulo}</p>
        </Fragment>
    );
}

//Validando los argumentos
PrimeraApp.propTypes={
    saludo: PropTypes.string.isRequired //Para determinar el tipo de valor
}

PrimeraApp.defaultProps={
    subtitulo:'Soy un subtitulo'
}


export default PrimeraApp;