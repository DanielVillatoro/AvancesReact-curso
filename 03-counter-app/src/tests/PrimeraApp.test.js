import React from 'react';
// import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import PrimeraApp from "../PrimeraApp";


describe('Pruebas en <PrimeraApp />',()=>{
    // test('Debe de mostrar el mensaje saludo "Hola, Daniel"', () => {
    //     const saludo='Hola, Daniel';
    //     const {getByText}=render(<PrimeraApp saludo={saludo}/>)
        
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('Debe mostrar <PrimeraApp /> correctamente', () => {
        const saludo='Hola, Daniel';
        const wrapper=shallow(<PrimeraApp saludo={saludo}/>);
        //toMatchSnapshot: 
        expect(wrapper).toMatchSnapshot();
    })
    

    
})