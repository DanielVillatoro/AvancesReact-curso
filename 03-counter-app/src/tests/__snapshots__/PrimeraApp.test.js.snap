exports[`Pruebas en <PrimeraApp /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PrimeraApp
    saludo="Hola, Daniel"
    subtitulo="Soy un subtitulo" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1>
          Hola, Daniel
</h1>,
        <p>
          Soy un subtitulo
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hola, Daniel",
        },
        "ref": null,
        "rendered": "Hola, Daniel",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Soy un subtitulo",
        },
        "ref": null,
        "rendered": "Soy un subtitulo",
        "type": "p",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1>
            Hola, Daniel
</h1>,
          <p>
            Soy un subtitulo
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hola, Daniel",
          },
          "ref": null,
          "rendered": "Hola, Daniel",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Soy un subtitulo",
          },
          "ref": null,
          "rendered": "Soy un subtitulo",
          "type": "p",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
